crates/consensus/auto-seal/tests/it/auto_mine.rs

```diff
@@ -0,0 +1,215 @@
+
+    // let mut file = tempfile::tempfile_in("./").unwrap();
+    // let chain_path = "./chain-spec";
+    let mut file = tempfile::NamedTempFile::new_in("./").unwrap();
+    
+    let chain_spec = custom_chain(&file);
+    write!(file, "{chain_spec:?}").unwrap();
+    let chain_path = file.path().to_str().unwrap();
+
+    println!("{:?}", std::fs::read_to_string(&chain_path).unwrap());
+    
+    let command = NodeCommand::<NoArgsCliExt<AutoMineConfig>>::parse_from([
+    // let mut command = NodeCommand::<()>::parse_from([
+        "reth",
+        "--dev",
+        "--datadir",
+        datadir,
+        "--debug.max-block",
+        "1",
+        "--chain",
+        chain_path,
+    ])
+    .with_ext::<NoArgsCliExt<AutoMineConfig>>(no_args);
+    // add custom chain
+    // command.chain = chain;
+
+    let runner = CliRunner::default();
+    runner.run_command_until_exit(|ctx| command.execute(ctx)).unwrap();
+    // let res = rx.try_recv().unwrap();
+
+}
+
+/// Build custom chain spec.
+fn build_chain_spec() -> Arc<ChainSpec> {
+    let chain = Chain::Id(2600);
+    let genesis = build_genesis();
+    let mut chain_spec = ChainSpecBuilder::default()
+        .chain(chain)
+        .cancun_activated()
+        .genesis(genesis)
+        .build();
+
+    let genesis_hash = chain_spec.genesis_hash();
+    chain_spec.genesis_hash = Some(genesis_hash);
+
+    Arc::new(chain_spec)
+}
+
+/// Build genesis for chain spec.
+fn build_genesis() -> Genesis {
+    // bob well-known private key: 0x99b3c12287537e38c90a9219d4cb074a89a16e9cdb20bf85728ebd97c343e342
+    let bob_address = hex!("6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b").into();
+    let bob_account = GenesisAccount::default().with_balance(U256::from(10_000));
+
+    let accounts: HashMap<Address, GenesisAccount> = HashMap::from([
+        (bob_address, bob_account),
+    ]);
+
+    Genesis::default().extend_accounts(accounts)
+}
+
+fn custom_chain(file: &NamedTempFile) {
+    let custom_genesis = r#"
+{
+    "nonce": "0x42",
+    "timestamp": "0x0",
+    "extraData": "0x5343",
+    "gasLimit": "0x1388",
+    "difficulty": "0x400000000",
+    "mixHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+    "coinbase": "0x0000000000000000000000000000000000000000",
+    "alloc": {
+        "0x6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b": {
+            "balance": "0x4a47e3c12448f4ad000000"
+        }
+    },
+    "number": "0x0",
+    "gasUsed": "0x0",
+    "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+    "config": {
+        "ethash": {},
+        "chainId": 2600,
+        "homesteadBlock": 0,
+        "eip150Block": 0,
+        "eip155Block": 0,
+        "eip158Block": 0,
+        "byzantiumBlock": 0,
+        "constantinopleBlock": 0,
+        "petersburgBlock": 0,
+        "istanbulBlock": 0,
+        "berlinBlock": 0,
+        "londonBlock": 0,
+        "terminalTotalDifficulty": 0,
```

Comment:
best to just construct the `Genesis` in memory, and use the struct to initialize the `ChainSpec`